runtime/java.go

```diff
@@ -36,13 +45,15 @@ func (conf javaRuntime) GetDependencies() []string {
 	return []string{"openjdk8-zulu-compact1"}
 }
 func (conf javaRuntime) Validate() error {
-	if conf.Base == "" {
+	// Only validate java-specific environment variables when base is openjdk-like.
+	if isCompatibleBase(conf.Base, javaPackages) {
```

Comment:
Could you please elaborate a bit more about why is this changed? Unfortunately, even from tests I am not sure I fully understand it.

I imagine I have java base package and then an HDFS package. The latter is used as the base for my own composed package, so I have e.g. `base:hadoop-hdfs:slave` and `runtime: java`. However, this change now prevents me from changing the `XMX` in the YAML?

Is this correct?

Reply:
Ok, I was missing the latter. As long as users are able to override, it is fine.